[Wallet] Redesign - Add header components

i1skn opened the pull request:
### Description
* add `TopBarIconButton` and `TopBarTextButton`.
* add generic `HeaderTitleWithSubtitle` 
* refresh header with balance

### Tested

Android emulator

### Backwards compatibility

Yes

### Screenshots
#### 1
![Screenshot_1589378972](https://user-images.githubusercontent.com/6160124/81826930-917fcb00-9538-11ea-8ada-db55b6f8f0c7.png)
#### 2
![Screenshot_1589379000](https://user-images.githubusercontent.com/6160124/81826919-8f1d7100-9538-11ea-8fb1-51af278f2b8b.png)
#### 3
![Screenshot_1589378963](https://user-images.githubusercontent.com/6160124/81826926-904e9e00-9538-11ea-9081-8ad5c9777152.png)
#### 4
![Screenshot_1589378978](https://user-images.githubusercontent.com/6160124/81826928-90e73480-9538-11ea-88bd-e48e313c164f.png)
#### 5 (when top bar button is pressed)
![Screenshot_1589459370](https://user-images.githubusercontent.com/6160124/81934608-86d43d00-95ef-11ea-8c28-99f011d5c29b.png)

i1skn committed:
Add header components

aaronmgdr commented on packages/mobile/src/components/BackButton.v2.tsx:
```diff
@@ -0,0 +1,8 @@
+import { navigateBack } from 'src/navigator/NavigationService'
+
+export default function MobileBackButton(props: { height?: number; color?: string }) {
+  const goBack = useCallback(() => navigateBack(), [navigateBack])
```
why not just pass navigateBack directly to BackButton?

aaronmgdr commented on packages/mobile/src/navigator/TopBarButton.v2.tsx:
```diff
@@ -1,31 +1,32 @@
-    >
-      {children}
-    </BorderlessButton>
+    <TouchableOpacity onPress={onPress} disabled={disabled} testID={testID}>
```
why this change? 

why remove Touchable? it handles having android phones have the nice ripple.

aaronmgdr commented on packages/react-components/components/BackButton.v2.tsx:
```diff
@@ -0,0 +1,35 @@
+
+const styles = StyleSheet.create({
+  container: {
+    paddingLeft: variables.contentPadding + 7,
```
im not sure about using these variables in new design. since everything is multiples of 8. the +7 worries me

jmrossy replied:
+1, setting up new variables may be nice.
But also, I find variables for things like padding to be unhelpful. How often will we say 'reduce padding across the app by 2'. Even if we did, unlikely that most things would line up right after the var change

i1skn replied:
I've moved this logic to `mobile` package. Regarding the padding itself I've changed it to `+6` (correlates with the design) to the standard padding we have.

<img width="655" alt="Screenshot 2020-05-14 at 12 56 26" src="https://user-images.githubusercontent.com/6160124/81958121-cb6ed100-960d-11ea-8f50-be20161a5b68.png">

aaronmgdr commented on packages/mobile/src/navigator/TopBarButton.v2.test.tsx:
```diff
@@ -5,7 +5,7 @@ import TopBarButton from 'src/navigator/TopBarButton.v2'
 describe('TopBarButton', () => {
   it('displays children', () => {
-    const { queryByText } = render(<TopBarButton>label</TopBarButton>)
+    const { queryByText } = render(<TopBarButton title={'label'} />)
```
hmm I think we should avoid changing interfaces

jmrossy replied:
I think it should be case-by-case. A title prop does make more sense to me than a child component

aaronmgdr commented on packages/mobile/src/navigator/TopBarButton.v2.tsx:
```diff
@@ -1,31 +1,32 @@
 
 const styles = StyleSheet.create({
   text: {
-    ...fontStyles.regular500,
```
hmm the buttons i reference in the header when i made this were def `500`

i1skn replied:
@coreycelo did some changes in the design, so I've reflected them here.

aaronmgdr replied:
ok cool

jmrossy commented on packages/react-components/icons/BackChevron.v2.tsx:
```diff
@@ -0,0 +1,18 @@
+  color: string
+}
+
+export default function BackChevron({ color, height }: Props) {
```
Setting default static props here like we do for some other icons may be a nice addition

jmrossy commented on packages/react-components/components/BackButton.v2.tsx:
```diff
@@ -0,0 +1,35 @@
+  return (
+    <View style={styles.container}>
+      <TouchableOpacity
+        hitSlop={variables.iconHitslop}
```
Maybe not a problem but just a note, I think hitslop only works if the parent container is sufficiently large (ie small parent will cut off hitslop area)

jmrossy commented on packages/mobile/src/navigator/Headers.v2.tsx:
```diff
@@ -0,0 +1,117 @@
+        available
+      </Trans>
+    ) : (
+      // TODO: a null balance doesn't necessarily mean it's loading
```
I find the possibility of null balance creates ugly conditionals all across the app and I wonder if we'd be better off just defaulting to 0. We'd lose a signal that says 'I don't know the balance' but it would simplify many places

i1skn replied:
I think, this is a bit out of the scope of this PR. But I agree, we should come up with a convenient approach how we represent unknown balance.

jmrossy reviewed (commented):
Mostly looks great!

i1skn committed:
PR comments

i1skn commented:
@aaronmgdr I've reworked `TopBarButton` and split it into `TopBarIconButton` and `TopBarTextButton`. I know that you have an opened PR, which relies on it and I'm happy to wait until we merge your one first.

i1skn committed:
Reflect TopBarButton logic in recently merged to master changes
